src/core_plugins/kibana/public/dashboard/directives/dashboard_panel_directive.js

```diff
@@ -0,0 +1,104 @@
+import _ from 'lodash';
+import 'ui/visualize';
+import 'ui/doc_table';
+import { loadPanelProvider } from 'plugins/kibana/dashboard/components/panel/lib/load_panel';
+import FilterManagerProvider from 'ui/filter_manager';
+import uiModules from 'ui/modules';
+import panelTemplate from 'plugins/kibana/dashboard/components/panel/panel.html';
+
+uiModules
+.get('app/dashboard')
+.directive('dashboardPanel', function (savedVisualizations, savedSearches, Notifier, Private, $injector) {
+  const loadPanel = Private(loadPanelProvider);
+  const filterManager = Private(FilterManagerProvider);
+
+  const services = require('plugins/kibana/management/saved_object_registry').all().map(function (serviceObj) {
+    const service = $injector.get(serviceObj.service);
+    return {
+      type: service.type,
+      name: serviceObj.service
+    };
+  });
+
+  /**
+   * Returns a unique id for storing the panel state in the persistent ui.
+   * @param panel
+   * @returns {string}
+   */
+  const getPersistedStateId = function (panel) {
+    return `P-${panel.panelId}`;
+  };
+
+  return {
+    restrict: 'E',
+    template: panelTemplate,
+    scope: {
+      /**
+       * Whether or not the dashboard this panel is contained on is in 'full screen mode'.
+       * @type {boolean}
+       */
+      isFullScreenMode: '=',
+      /**
+       * The parent's persisted state is used to create a child persisted state for the
+       * panel.
+       * @type {PersistedState}
+       */
+      parentUiState: '=',
```

Comment:
The `parentUiState` couples this to the parent quite tightly, because there is no limitation to how it can be manipulated. See the overall review comment for more thoughts.